### PR TITLE
Zero Supply Liquid Token

### PR DESCRIPTION
**Pull request by barakman**
1. When purchasing at supply = 0, return the reserve amount without fee.
2. When selling the entire supply, return the entire reserve without fee.

### CODE REVIEW

File: solidity/contracts/converter/BancorConverter.sol [hunk 1]
```diff
@@ -507,8 +507,13 @@ contract BancorConverter is IBancorConverter, TokenHandler, SmartTokenController
         validReserve(_reserveToken)
         returns (uint256, uint256)
     {
+        uint256 supply = token.totalSupply();
+
+        if (supply == 0)
+            return (_depositAmount, _depositAmount);
```
Comment (yudilevi) [inline solidity/contracts/converter/BancorConverter.sol:513]: I'm pretty sure the 2nd value here should be 0 (fee), no?

File: solidity/contracts/converter/BancorConverter.sol [hunk 2]
```diff
@@ -536,9 +541,15 @@ contract BancorConverter is IBancorConverter, TokenHandler, SmartTokenController
         validReserve(_reserveToken)
         returns (uint256, uint256)
     {
+        uint256 supply = token.totalSupply();
+        uint256 balance = reserveBalance(_reserveToken);
+
+        if (supply == _sellAmount)
+            return (balance, balance);
```
Comment (yudilevi) [inline solidity/contracts/converter/BancorConverter.sol:548]: same here - 2nd value should be 0, no?

File: solidity/contracts/converter/BancorConverter.sol (hunk 1 above)
Comment (barakman) [reply solidity/contracts/converter/BancorConverter.sol:513]: @yudilevi : No, it should be amount minus fee.

File: solidity/contracts/converter/BancorConverter.sol (hunk 2 above)
Comment (barakman) [reply solidity/contracts/converter/BancorConverter.sol:548]: @yudilevi : No, it should be amount minus fee.

File: solidity/contracts/converter/BancorConverter.sol (hunk 1 above)
Comment (barakman) [reply solidity/contracts/converter/BancorConverter.sol:513]: @yudilevi : Damn, you're right!

File: solidity/contracts/converter/BancorConverter.sol (hunk 2 above)
Comment (barakman) [reply solidity/contracts/converter/BancorConverter.sol:548]: @yudilevi : Damn, you're right!

Comment (barakman) [reply solidity/contracts/converter/BancorConverter.sol]: @yudilevi : Fixed.

File: solidity/contracts/converter/BancorConverter.sol (hunk 1 above)
Comment (barakman) [reply solidity/contracts/converter/BancorConverter.sol]: @yudilevi : Fixed.

**Pull request merged**